src/plugins/intel_cpu/src/emitters/snippets/cpu_kernel_executor_table.hpp

```diff
@@ -16,28 +16,24 @@ class CPUKernelExecutor : public snippets::KernelExecutor<Conf, KernelType> {
      CPUKernelExecutor(ov::intel_cpu::MultiCacheWeakPtr kernel_cache, std::shared_ptr<Conf> c) :
                        snippets::KernelExecutor<Conf, KernelType>(c), m_kernel_cache(std::move(kernel_cache)) {}
      struct Key {
-         explicit Key(std::shared_ptr<Conf> c) : config{std::move(c)} {}
-         const std::shared_ptr<Conf> config;
+         explicit Key(const std::shared_ptr<const Conf>& c) : config{c} {}
+         const std::shared_ptr<const  Conf> config;
          size_t hash() const { return config->hash(); }
          bool operator==(const Key& rhs) const { return *config == *rhs.config; }
      };
-    void update_kernel() override {
-        OPENVINO_ASSERT(m_config && m_config->is_completed(), "Update kernel was called with invalid config");
+    void update_kernel(const std::shared_ptr<const Conf>& config, std::shared_ptr<KernelType>& kernel) const override {
         const auto& cache = m_kernel_cache.lock();
         OPENVINO_ASSERT(cache, "Invalid kernel cache pointer in CPUKernelExecutor::update_kernel()");
-        const auto& lookup_result = cache->getOrCreate(Key(m_config),
+        const auto& lookup_result = cache->getOrCreate(Key(config),
                                                        [this](const Key& k) {
                                                             return compile_kernel(k.config);
                                                        });
-        m_kernel = lookup_result.first;
-        OPENVINO_ASSERT(m_kernel, "Failed to compile kernel executor");
```

Comment:
We really don't have to because this check if performed in `KernelExecutor::update_by_*`
[here](https://github.com/IvanNovoselov/openvino/blob/6583eaaa63ca4d06f7e3f3f9c4685eac5901e4a7/src/common/snippets/include/snippets/kernel_executor_table.hpp#L76)